test/AutoRest.TestServerLowLevel.Tests/body-complex.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Xml;
+using AutoRest.TestServer.Tests.Infrastructure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using body_complex;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class BodyComplexTest: TestServerLowLevelTestBase
+    {
+        public BodyComplexTest(TestServerVersion version) : base(version) { }
+
+        [Test]
+        public Task GetComplexBasicValid() => Test(async (host, key) =>
+        {
+            var result = await new BasicClient(key, host).GetValidAsync();
+            var responseBody = JsonData.FromBytes(result.Content.ToMemory());
+            Assert.AreEqual("abc", (string)responseBody["name"]);
+            Assert.AreEqual(2, (int)responseBody["id"]);
+            Assert.AreEqual("YELLOW", (string)responseBody["color"]);
+        });
+
+        [Test]
+        public Task PutComplexBasicValid() => TestStatus(async (host, key) =>
+        {
+            var data = new JsonData(new Dictionary<string, object> {
+                ["name"] = "abc",
+                ["id"] = 2,
+                ["color"] = "Magenta"
+            });
+            return await new BasicClient(key, host).PutValidAsync(RequestContent.Create(data));
```

Comment:
I think we don't need to pass the key every time. We can add a static Key on the TestServerLowLevelTestBase